fix(ast): estree compat `AssignmentTargetPropertyIdentifier`

hi-ogawa opened the pull request:
This one seemed less straight forward than `AssignmentTargetPropertyProperty`, so I split it from https://github.com/oxc-project/oxc/pull/9005. This requires somewhat redundant custom serialization logic, but this is all I came up with. I'm curious to know if there's any good shortcut for this.

hi-ogawa commented:
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9006?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#9006** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9006?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9006?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#9005** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9005?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#8998** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/8998?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=9006).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

overlookmotel reviewed (approved):
Yes, I think there's a simpler way to implement this. Merging this now, and I'll do a follow-up PR.

overlookmotel commented:
By the way, it's been a bit of a struggle to get all these PRs merged, due to inadequacies in Graphite and the merge conflicts in the snapshot file.

One thing that would help is if you'd be able to squash all PRs down to a single commit each please. Multiple commits should be no problem, but squashing seems to confuse Graphite less.

Also, please feel free to merge your own PRs in this area of work. If it passes more tests, it's good! I can always follow up after they're merged if I want to make tweaks or query anything.

hi-ogawa committed:
fix(ast): estree compat `AssignmentTargetPropertyIdentifier` (#9006)

This one seemed less straight forward than `AssignmentTargetPropertyProperty`, so I split it from https://github.com/oxc-project/oxc/pull/9005. This requires somewhat redundant custom serialization logic, but this is all I came up with. I'm curious to know if there's any good shortcut for this.

hi-ogawa commented:
> @hi-ogawa By the way, it's been a bit of a struggle to get all these PRs merged, due to inadequacies in Graphite and the merge conflicts in the snapshot file.
> 
> One thing that would help is if you'd be able to squash all PRs down to a single commit each please. Multiple commits should be no problem, but squashing seems to confuse Graphite less.
> 
> Also, please feel free to merge your own PRs in this area of work. If it passes more tests, it's good! I can always follow up after they're merged if I want to make tweaks or query anything.

@overlookmotel Understood! Sorry about the conflict and thanks for taking care of them :pray: I'll try to proceed my own for easier merge.
